ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/KafkaTopicInfoExtended.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import java.util.List;
+import java.util.Objects;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonSubTypes({})
+public class KafkaTopicInfoExtended {
```

Comment:
thoughts on adding a boolean for `extended` to the original entity? That way we can avoid essentially duplicating a lot of logic in `KafkaTopicInfo`, `KafkaTopicListExtended` and `KafkaTopicInfoTableBuilder`

Reply:
I definitely understand the urge to collapse them. Here are a few counter arguments, none of which I'm very attached to:

1. It allows us to let the two structures vary independently.
2. It somewhat follows a convention I saw with SourceDescription and SourceInfo::Table (maybe for reason 1)?
3. It's very clear what is in the entity, which potentially saves us branching and special case code, e.g. introducing Optionals for the "simple" case.

We could, for the moment, compose KafkaTopicInfo in KafkaTopicInfoExtended and re-export the common fields name, replicateInfo, but that doesn't buy us too much and I don't think it gets you the level of unification you're looking for.

Ultimately, I'm happy to follow team convention. If unification looks more like the rest of the code base let me know and I will make it happen.

Reply:
Funny that you mention `SourceDescription` - I looked at that before suggesting this. You can see it has a `extended` boolean that it leverages in `Console` to determine how it prints the response.

Either way, I'm not attached either to having them be a single entity. Really the important part to me is that the duplicated business logic is shared (e.g. `filterKsqlInternalTopics` and `getTopicReplicaInfo`)

Reply:
Thanks @agavra! Totally agree on the business logic. I'll extract those where they're duplicated and follow up.